app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -218,7 +218,27 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
         )
     }
 
-    private fun showPinDialogWarning(newValue: Boolean) {
+    private fun CheckBoxPreference.updateSyncEngineStateWithPinWarning(newValue: Boolean) {
+        val manager =
+            activity?.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
+        if (manager.isKeyguardSecure
+            || !newValue
+            || !requireContext().settings().shouldShowSecurityPinWarningSync
+        ) {
+            updateSyncEngineState(context, SyncEngine.CreditCards, newValue)
+        } else {
+            showPinDialogWarning(SyncEngine.CreditCards, newValue)
+        }
+    }
+
+    private fun updateSyncEngineState(context: Context, engine: SyncEngine, newState: Boolean) {
```

Comment:
moved from above (line 167)

Reply:
I can imagine people being worried about losing the initial commit history associated with this. To mitigate that concern, I would prefer we also improve the code quality by adding a kDoc with its @param to ensure the loss of history is worthwhile. I would also just check the git blame to see if this function was part of some bigger feature. If it is, I would just note that in the PR or issue somewhere in case, anyone ever wants to refer back to it or weigh whether or not it is worthwhile to make this change.